ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -48,33 +49,63 @@ import kotlin.collections.component2
 import kotlin.collections.set
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.reactive.asFlow
 
 /** View model for the Data Collection fragment. */
+@OptIn(ExperimentalCoroutinesApi::class)
 @HiltViewModel
 class DataCollectionViewModel
 @Inject
 internal constructor(
   private val viewModelFactory: ViewModelFactory,
   private val submissionRepository: SubmissionRepository,
+  private val locationOfInterestRepository: LocationOfInterestRepository,
   private val locationOfInterestHelper: LocationOfInterestHelper,
   private val popups: Provider<EphemeralPopups>,
   private val navigator: Navigator,
   @ApplicationScope private val externalScope: CoroutineScope,
   @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
-  private val savedStateHandle: SavedStateHandle
+  private val savedStateHandle: SavedStateHandle,
+  surveyRepository: SurveyRepository,
 ) : AbstractViewModel() {
 
-  val submission: @Hot(replays = true) LiveData<Submission>
-  val jobName: @Hot(replays = true) LiveData<String>
-  val loiName: @Hot(replays = true) LiveData<String>
+  // TODO(#1541): Set loiId once Suggest LOI task is completed.
+  private val loiId: StateFlow<String?> = MutableStateFlow(savedStateHandle["locationOfInterestId"])
+  private val activeSurvey: Survey = requireNotNull(surveyRepository.activeSurvey)
+  private val job: Job =
+    activeSurvey.getJob(requireNotNull(savedStateHandle["jobId"])).orElseThrow()
```

Comment:
I feel that the previous approach of using generated keys instead of using hard-coded string values is less error prone. Is there a way to share the key values between here and the nav_graph?

Reply:
I don't have any experience with using `SavedStateHandle`. Will defer to your final call here as well.

Reply:
ACK, will leave as is and we can revisit if it becomes an issue.